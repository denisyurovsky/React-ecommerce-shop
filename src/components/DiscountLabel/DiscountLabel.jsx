import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';

import { formatDiscountInPercents } from '../../helpers/utils/formatData';

import styles from './DiscountLabel.module.scss';

export const DiscountLabel = ({ price, discountPrice }) => {
  return (
    <Typography className={styles.label}>
      {formatDiscountInPercents(price, discountPrice)}
    </Typography>
  );
};

DiscountLabel.propTypes = {
  price: PropTypes.number,
  discountPrice: PropTypes.number,
};
