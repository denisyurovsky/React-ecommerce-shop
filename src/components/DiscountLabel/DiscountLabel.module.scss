@import "../../assets/styles/colors.scss";

.label {
  padding: 4px 8px;
  color: $greyLight;
  background-color: $red700;
  border-radius: 20px;
}
