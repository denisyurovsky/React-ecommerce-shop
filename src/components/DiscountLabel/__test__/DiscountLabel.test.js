import { render } from '@testing-library/react';
import React from 'react';

import { DiscountLabel } from '../DiscountLabel';

describe('DiscountLabel component', () => {
  describe('snapshots', () => {
    it('renders a snapshot with two prices', () => {
      const { asFragment } = render(
        <DiscountLabel price={100} discountPrice={10} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
