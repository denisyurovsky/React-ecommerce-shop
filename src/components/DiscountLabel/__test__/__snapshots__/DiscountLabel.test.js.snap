// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountLabel component snapshots renders a snapshot with two prices 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 label css-ahj2mt-MuiTypography-root"
  >
    90 %
  </p>
</DocumentFragment>
`;
